presto-parquet/src/main/java/io/prestosql/parquet/reader/ShortDecimalColumnReader.java

```diff
@@ -44,7 +46,13 @@ else if (columnDescriptor.getPrimitiveType().getPrimitiveTypeName() == INT64) {
             else {
                 decimalValue = getShortDecimalValue(valuesReader.readBytes().getBytes());
             }
-            type.writeLong(blockBuilder, decimalValue);
+
+            if (isShortDecimal(type)) {
```

Comment:
This check is only needed for the last branch above, no? If so, I'd change it to do this, instead:

```
if (xxx = INT32) {
    ...
    type.writeLong(blockBuilder, decimalValue);
}
else if (xxx = INT64) {
    ...
    type.writeLong(blockBuilder, decimalValue);
}
else {
    ....
    if (isShortDecimal(type)) {
        type.writeLong(...);
    }
    else {
        type.writeSlice(...)
    }
}